easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -472,47 +472,57 @@ def dump(self, fp):
-                    return '"%s"' % x
-                else:
-                    return "'%s'" % x
-            else:
-                return "%s" % x
-
         # ordered groups of keys to obtain a nice looking easyconfig file
         grouped_keys = [
+            ['easyblock'],
             ['name', 'version', 'versionprefix', 'versionsuffix'],
             ['homepage', 'description'],
             ['toolchain', 'toolchainopts'],
-            ['source_urls', 'sources'],
+            ['sources', 'source_urls'],
             ['patches'],
             ['builddependencies', 'dependencies', 'hiddendependencies'],
+            ['osdependencies'],
+            ['preconfigopts', 'configopts'],
+            ['prebuildopts', 'buildopts'],
+            ['preinstallopts', 'installopts'],
             ['parallel', 'maxparallel'],
-            ['osdependencies']
         ]
 
+        last_keys = [
+            ['sanity_check_paths'],
+            ['moduleclass'],
+        ]
+
+        def check_and_print(keyset):
+            """
+            Internal function checking for default values
+            """
+            for group in keyset:
+                printed = False
+                for key1 in group:
+                    val = self._config[key1][0]
+                    for key2, [def_val, _, _] in DEFAULT_CONFIG.items():
+                        # only print parameters that are different from the default value
+                        if key1 == key2 and val != def_val:
+                            ebtxt.append("%s = %s" % (key1, quote_str(val, escape_newline=True)))
+                            printed_keys.append(key1)
+                            printed = True
+                if printed:
+                    ebtxt.append("")
+
+
         # print easyconfig parameters ordered and in groups specified above
         ebtxt = []
         printed_keys = []
-        for group in grouped_keys:
-            for key1 in group:
-                val = self._config[key1][0]
-                for key2, [def_val, _, _] in DEFAULT_CONFIG.items():
-                    # only print parameters that are different from the default value
-                    if key1 == key2 and val != def_val:
-                        ebtxt.append("%s = %s" % (key1, to_str(val)))
-                        printed_keys.append(key1)
-            ebtxt.append("")
+        check_and_print(grouped_keys)
 
         # print other easyconfig parameters at the end
         for key, [val, _, _] in DEFAULT_CONFIG.items():
-            if not key in printed_keys and val != self._config[key][0]:
-                ebtxt.append("%s = %s" % (key, to_str(self._config[key][0])))
+            if not key in printed_keys and not key in [k for sublist in last_keys for k in sublist] and val != self._config[key][0]:
+                ebtxt.append("%s = %s" % (key, quote_str(self._config[key][0], escape_newline=True)))
+
+        # print last two parameters
+        check_and_print(last_keys)
```

Comment:
maybe make `last_keys` a flat list, and change this to

```python
check_and_print([[k] for k in last_keys])
```

Just something that popped to mind, `last_keys` isn't going to change most likely?